shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java

```diff
@@ -0,0 +1,118 @@
+    
+    private static final String ADD_RESOURCE = "ADD RESOURCE";
+    
+    private static final String RESOURCES = " %s ("
+                                            + System.lineSeparator()
+                                            + "\tURL=\"%s\","
+                                            + System.lineSeparator()
+                                            + "\tUSER=%s,"
+                                            + System.lineSeparator()
+                                            + "\tPASSWORD=%s,"
+                                            + System.lineSeparator()
+                                            + "\tPROPERTIES(%s%s%s%s%s)"
+                                            + System.lineSeparator()
+                                            + "),";
+    
+    @Override
+    protected Collection<String> getColumnNames() {
+        return Collections.singleton("converted DistSQL");
+    }
+    
+    @Override
+    protected Collection<LocalDataQueryResultRow> getRows(final ContextManager contextManager) {
+        File file = new File(getSqlStatement().getFilePath());
+        YamlProxyDatabaseConfiguration yamlConfig;
+        try {
+            yamlConfig = YamlEngine.unmarshal(file, YamlProxyDatabaseConfiguration.class);
+        } catch (final IOException ex) {
+            throw new ShardingSphereException(ex);
+        }
+        String convertedDistSQL = generateConvertedDistSQL(yamlConfig);
+        return Collections.singleton(new LocalDataQueryResultRow(convertedDistSQL));
+    }
+    
+    private String generateConvertedDistSQL(final YamlProxyDatabaseConfiguration yamlConfig) {
+        StringBuilder convetedDistSQL = new StringBuilder();
+        appendCreateDatabaseDistSQL(yamlConfig.getDatabaseName(), convetedDistSQL);
+        appendAddResourceDistSQL(yamlConfig.getDataSources(), convetedDistSQL);
+        return convetedDistSQL.toString();
+    }
+    
+    private void appendCreateDatabaseDistSQL(final String databaseName, final StringBuilder stringBuilder) {
+        if (databaseName.isEmpty()) {
+            return;
+        }
+        stringBuilder.append(String.format(CREATE_DATABASE, databaseName));
+    }
+    
+    private void appendAddResourceDistSQL(final Map<String, YamlProxyDataSourceConfiguration> databaseResource, final StringBuilder stringBuilder) {
+        if (databaseResource.isEmpty()) {
+            return;
+        }
+        if (null == stringBuilder) {
+            stringBuilder.append(ADD_RESOURCE);
+        } else {
+            stringBuilder.append(String.format(System.lineSeparator()
+                                               + System.lineSeparator()
+                                               + ADD_RESOURCE));
+        }
+        stringBuilder.append(ADD_RESOURCE);
+        for (Map.Entry<String, YamlProxyDataSourceConfiguration> entry : databaseResource.entrySet()) {
+            String resourceName = entry.getKey();
```

Comment:
There are too many `entry.getValue()` in the following code, get a dataSourceConfiguration object to use.